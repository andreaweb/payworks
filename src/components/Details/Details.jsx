import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Results from '../Results/Results';
import ErrorAlert from '../ErrorAlert/ErrorAlert';
import Button from '@material-ui/core/Button';
import { Link } from 'react-router-dom';
import { MuiThemeProvider } from '@material-ui/core/styles';
import { theme } from '../../common/MuiTheme.js';
import './Details.scss';

const rows = [
  { id: 'branch', numeric: false, disablePadding: false, label: 'Branch Name' },
  { id: 'sha', numeric: false, disablePadding: false, label: 'Sha' },
  { id: 'url', numeric: false, disablePadding: false, label: 'Link' }
];

class Details extends Component {
	state = {
		branches: null,
		error: false
	};
	componentDidMount(){
		const repoName = this.props.match.params.name;
		const orgName = this.props.match.params.org;
		let fetchURL = `https://api.github.com/repos/${orgName}/${repoName}/branches`;
		fetch(fetchURL)
		.then(res => {
      console.log(res);
      if(res.ok){
        return res.json();
      }else{
        return null; //render an error inside Results
      }
    })
		.then(data => {
			console.log(data);
			let branches = data;
			let arr = [];
			branches.map(function(branch){
				const arrItem = {
					branch: branch.name,
					sha: branch.commit.sha,
					url: <a href={branch.commit.url}>Redirect to API</a>
				};
				arr.push(arrItem);
			});
			return arr;
		})
		.then(arr =>{
			this.setState({branches: arr, error: false});
		})
		.catch(err =>{
			this.setState({error: true, errorMsg: String(err)});
		});
	}
	render() {
		return (
			<main>
				<div className="row row--details">
					<span>
						<span className="lighter">Showing repositories of </span>
						<span className="company">
							{this.props.match.params.org}/
							{this.props.match.params.name} 
						</span>
					</span>
					<MuiThemeProvider theme={theme}>
						<Button variant="outlined" color="primary">
							<Link to="/" className="button-link">
								Back to Repositories
							</Link>
						</Button>
					</MuiThemeProvider>
				</div>
				{this.state.error && 
					<ErrorAlert 
						resetError={this.props.resetError} 
						error={this.state.error}
						errorMsg={`${this.state.errorMsg}, redirecting to home...`}
						redirect={true}
						history={this.props.history}
					/>
				}
				{this.state.branches &&
					<Results 
						arr={this.state.branches} 
						openModal={this.props.openModal}
						rows={rows}
					/>
				}
			</main>
		);
	}
}

Details.propTypes = {
	match: PropTypes.object,
	history: PropTypes.object,
	error: PropTypes.bool,
	errorMsg: PropTypes.string,
	resetError: PropTypes.func,
	openModal: PropTypes.func
};

export default Details;
