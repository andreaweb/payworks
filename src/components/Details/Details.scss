.button-link{
	text-decoration: none;
	color: #fff;
}
.row--details{
	margin: 2vh 0 3vh;
	color: #fff;
	font-size: 1.5em;
}