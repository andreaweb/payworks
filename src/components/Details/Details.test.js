import React from 'react';
import ReactDOM from 'react-dom';
import Details from './Details';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

it('renders correctly', () => {
    const tree = renderer.create(
    	<Router><Details match={{params: {org: 'payworks', name: 'slate'}}} /></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});