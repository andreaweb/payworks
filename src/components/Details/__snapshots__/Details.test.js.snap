// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <div
    className="row row--details"
  >
    <span>
      <span
        className="lighter"
      >
        Showing repositories of 
      </span>
      <span
        className="company"
      >
        payworks
        /
        slate
      </span>
    </span>
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-outlined-9 MuiButton-outlinedPrimary-10"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        <a
          className="button-link"
          href="/payworks"
          onClick={[Function]}
        >
          Back to Repositories
        </a>
      </span>
      <span
        className="MuiTouchRipple-root-40"
      />
    </button>
  </div>
  <div
    className="MuiCircularProgress-root-30 MuiCircularProgress-colorPrimary-33 MuiCircularProgress-indeterminate-32 spinner"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-35"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-36 MuiCircularProgress-circleIndeterminate-38"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</main>
`;
